src/libostree/ostree-repo.c

```diff
@@ -2771,18 +2784,22 @@ reload_core_config (OstreeRepo          *self,
       self->zlib_compression_level = OSTREE_ARCHIVE_DEFAULT_COMPRESSION_LEVEL;
   }
 
-  { g_autofree char *min_free_space_percent_str = NULL;
+  { g_autofree char *min_free_space_str = NULL;
+    self->min_free_space_str = NULL;
+
     /* If changing this, be sure to change the man page too */
-    const char *default_min_free_space = "3";
+    const char *default_min_free_space = "3%";
```

Comment:
Well...but we have a backcompat hazard here as we didn't previously require `%`.  So perhaps it is best to introduce a new config option.

Reply:
Or interpret a bare number as a percentage for the back compat case?

Reply:
@ramcq Won't it give a wrong idea? For e.g.
(i) `min-free-space-percent=3`
(ii) `min-free-space-percent=150MB`
or am I missing something?

I wonder if adding an alias and introducing new config option  are same thing? Do we have an exisitng alias case which I can look-up?

Reply:
>If we have two options, an interesting question arises if we want to support setting both simultaneously - I'd be fine trying to do so, I'd also be fine making it an error for now.

I think, making it an error sounds logical to me here.


>I want to be clear here: Is this just an idea of yours or are you planning to actually ship a non-zero min-free-space in EndlessOS?

It's not just an idea but a step in direction of https://github.com/ostreedev/ostree/pull/1602#issuecomment-393289405 . Also, min-free-space (% or specific size) should guard against possibility of hitting `ENOSPC`. So yes, Endless would be very much interesting in this as it provides a bit more fine tuning of the config option.

Reply:
Endless certainly wants to ship something that does this kind of thing.

Reply:
@cgwalters  Quick question: If both `min-free-space` and `min-free-space-percent` are specified in the config file(for some odd reasons), do we error that out, or treat one of them as default (say `min-free-space`) and proceed ahead with discarding the other one ?